src/fund.rs

```diff
@@ -0,0 +1,63 @@
+use ethereum_types::{Address, H256, U256};
+use ethcore::{self, state::{State, CleanupMode}};
+use oasis_ethwasi_runtime_common::parity::NullBackend;
+extern crate chrono;
+use chrono::{DateTime, NaiveDateTime, Utc, Datelike};
+extern crate date_time;
+use date_time::month_tuple::MonthTuple;
+use std::str::FromStr;
+use std::convert::TryInto;
+use byteorder::{BigEndian, ReadBytesExt};
+
+fn timestamp_to_mt(timestamp:i64) -> MonthTuple{
+  let dt = DateTime::<Utc>::from_utc(NaiveDateTime::from_timestamp(timestamp, 0), Utc);
+  return MonthTuple::new(dt.year().try_into().unwrap(),
+                         dt.month().try_into().unwrap()).unwrap();
+}
+
+pub struct FundManager;
+
+impl FundManager {
+
+    /// Unlock address.
+    const BENEFICIARY: &'static str = "7110316b618d20d0c44728ac2a3d683536ea682b";
+    /// Adjust period.
+    const PERIOD: u32 = 20;
+    // Factor
+    const FACTOR: f32 = 0.5;
+    /// Total Amount.
+    const TOTAL_AMOUNT: u32 = 21000000;
+    /// Initial unlock bucket.
+    const INITIAL_BUCKET: f32 = 525000.0;
+
+    pub fn try_unlock(timestamp:u64, state:&mut State<NullBackend>) {
```

Comment:
Should we make this function atomic?
Once it failed, we should rollback states.

Reply:
I think this case should be covered by oasis-parity state.commit function (I saw oasis-parity use storage_cache internal) and consensus layer.
If some compute node fail within execution then its' state change should not be acceptable by others and need drop & re-sync.
Do my opinion is reasonable?

Reply:
It makes sense.